streamz/sources.py

```diff
@@ -221,21 +221,26 @@ def start(self):
             def close(ref):
                 ob = ref()
                 if ob is not None and ob.consumer is not None:
-                    ob.consumer.close()
+                    consumer = ob.consumer
+                    ob.consumer = None
+                    consumer.unsubscribe()
+                    consumer.close()  # may raise with latest ck, that's OK
```

Comment:
Can you briefly explain a bit more about what is going on here?  What was failing before?  Should any of this go within finalize calls or try/finally blocks?  Should we be using thread locks around any of this?

Reply:
> Is it safe mixing a thread lock with a finalize call? I'm not sure where you are suggesting putting in a try.

I didn't mean to suggest anything concrete, I was just listing various mechanisms that are sometimes used in situations like this.  

OK, so I understand now why we set `ob.consumer = None` when we close.  

Under what circumstances does this raise?  Should we be catching those?